components/shared/ui/menu.tsx

```diff
@@ -29,9 +30,24 @@ const isCurrentNavigationItemActive = (navigation: Block_Navigation, router: Nex
   const pathWithoutQuerystring = router.asPath.replace(/\?.*/, '');
   const pathSegments = pathWithoutQuerystring.split("/");
   const topLevelSegment = pathSegments[1];
-  return (navigation.elements.pageLink.linkedItems[0]?.elements.url.value === topLevelSegment);
+  const pageLink = navigation.elements.pageLink.linkedItems[0];
+  return (pageLink?.system.codename === contentTypes.page.codename && pageLink.elements.url.value === topLevelSegment);
```

Comment:
This doesn't seem to work. The current page is not highlighted in the menu. 
![image](https://github.com/kontent-ai-bot/kontent-ai-new-sample-app/assets/25480503/38d68f33-0a29-46ff-82a2-6f12b31e8791)

Reply:
fixed in 041c43d